### PR TITLE
feat: InMemoryRandomAccessStream

### PR DESCRIPTION
**Pull request by workgroupengineering**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

- Feature

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UWP/Storage/Streams/InMemoryRandomAccessStream.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+﻿using System.IO;
```
Comment (jeromelaban) [inline src/Uno.UWP/Storage/Streams/InMemoryRandomAccessStream.cs:1]: #nullable enable

```suggestion
﻿using System.IO;
```

File: src/Uno.UWP/Storage/Streams/InMemoryRandomAccessStream.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+namespace Windows.Storage.Streams
+{
+	public partial class InMemoryRandomAccessStream: IStreamWrapper
+	{
+		private MemoryStream _stream;
```
Comment (jeromelaban) [inline src/Uno.UWP/Storage/Streams/InMemoryRandomAccessStream.cs:7]:
```suggestion
		private readonly MemoryStream _stream;
```

**Review by jeromelaban — COMMENTED** [review]
Thanks for your contribution! Could you also add some unit or runtime tests ?

File: src/Uno.UWP/Storage/Streams/InMemoryRandomAccessStream.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+			_stream.WriteAsyncOperation(buffer);
+
+		public IAsyncOperation<bool> FlushAsync() =>
+			_stream.FlushAsyncOperation();
+
+		public IInputStream GetInputStreamAt(ulong position)
+		{
+			if (!CanRead)
+			{
+				throw new NotSupportedException("The file has not been opened for read.");
```
Comment (MartinZikmund) [inline src/Uno.UWP/Storage/Streams/InMemoryRandomAccessStream.cs:56]: I would change this message to omit "file" as the stream source may be something else

File: src/Uno.UWP/Storage/Streams/InMemoryRandomAccessStream.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+			_stream.Seek((long)position, SeekOrigin.Begin);
+
+			return new InputStreamOverStream(_stream);
+		}
+
+		public IOutputStream GetOutputStreamAt(ulong position)
+		{
+			if (!CanWrite)
+			{
+				throw new NotSupportedException("The file has not been opened for write.");
```
Comment (MartinZikmund) [inline src/Uno.UWP/Storage/Streams/InMemoryRandomAccessStream.cs:68]: Same here

File: src/Uno.UWP/Storage/Streams/InMemoryRandomAccessStream.cs [hunk 5]
```diff
@@ -0,0 +1,76 @@
+﻿#nullable enable
+using System;
+using System.IO;
+using Windows.Foundation;
+
+namespace Windows.Storage.Streams
```
Comment (MartinZikmund) [inline src/Uno.UWP/Storage/Streams/InMemoryRandomAccessStream.cs:6]: Use file-scoped namespaces for new files 👍

**Review by MartinZikmund — APPROVED** [review]

**Pull request merged**